harold/plugins/deploy.py

```diff
@@ -234,17 +307,48 @@ def update_conch(self, irc):
             new_conch = None
         self.current_conch = new_conch
 
-    def hold(self, irc, reason):
+    def hold(self, irc, type, reason):
+        """
+        Sets a deploy hold status on a salon
+
+        :param irc: IrcPlugin object for sending IRC messages
+        :param type: DeployHoldType enum
+        :param reason: Reason text
+
+        :return: None
+        """
         if not reason:
             reason_text = "no reason given"
         else:
             reason_text = " ".join(reason)
+
+        # If the type is a manual hold and we are currently in a
+        # code freeze then there is likely some emergency that
+        # needs to be addressed.  However, once that hold has been
+        # lifted we wish to restore the previous freeze.  Thus we
+        # need to preserve the freeze reason for restoration.
+        if type == DeployHoldType.manual and self.current_hold_type == DeployHoldType.code_freeze:
+            self.previous_freeze = self.current_hold
```

Comment:
:no_bell: As noted offline, it would be nice to persist `previous_freeze` to the database in case harold crashes.

Reply:
Yup.  I can either update the DB or just have the Google Apps script trigger a reminder every day.

Reply:
Probably better to do the reminder every day. DB is a bit funky right now.